DMX.Core.Api.Tests.Unit/Services/Orchestrations/LabOrchestrationServiceTests.Exceptions.RetrieveById.cs

```diff
@@ -0,0 +1,147 @@
+
+            LabOrchestrationDependencyValidationException
+                actualLabOrchestrationDependencyValidationException =
+                    await Assert.ThrowsAsync<LabOrchestrationDependencyValidationException>(
+                        actualLabTask.AsTask);
+
+            // then
+            actualLabOrchestrationDependencyValidationException.Should()
+                .BeEquivalentTo(expectedLabOrchestrationDependencyValidationException);
+
+            this.labServiceMock.Verify(service =>
+                service.RetrieveLabByIdAsync(It.IsAny<Guid>()),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+                    expectedLabOrchestrationDependencyValidationException))),
+                        Times.Once);
+
+            this.labServiceMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.externalLabServiceMock.VerifyNoOtherCalls();
+        }
+
+        [Theory]
+        [MemberData(nameof(LabDependencyExceptions))]
+        public async Task ShouldThrowOrchestrationDependencyExceptionOnRetrieveByIdDependencyErrorOccursAndLogItAsync(
+            Xeption dependencyException)
+        {
+            // given
+            Guid someLabId = Guid.NewGuid();
+
+            var expectedLabOrchestrationDependencyException =
+                new LabOrchestrationDependencyException(
+                    dependencyException.InnerException as Xeption);
+
+            this.labServiceMock.Setup(service =>
+                service.RetrieveLabByIdAsync(It.IsAny<Guid>()))
+                    .ThrowsAsync(dependencyException);
+
+            // when
+            ValueTask<Lab> retrieveLabTask =
+                this.labOrchestrationService.RetrieveLabByIdAsync(someLabId);
+
+            LabOrchestrationDependencyException actualLabOrchestrationDependencyException =
+                await Assert.ThrowsAsync<LabOrchestrationDependencyException>(
+                    retrieveLabTask.AsTask);
+
+            // then
+            actualLabOrchestrationDependencyException.Should()
+                .BeEquivalentTo(expectedLabOrchestrationDependencyException);
+
+            this.labServiceMock.Verify(service =>
+                service.RetrieveLabByIdAsync(someLabId),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+                    expectedLabOrchestrationDependencyException))),
+                        Times.Once);
+
+            this.labServiceMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.externalLabServiceMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowOrchestrationServiceExceptionOnRetrieveByIdIfErrorOccursAndLogItAsync()

```

Comment:
Rename this to: `ShouldThrowServiceExceptionOnRetrieveByIdIfServiceErrorOccursAndLogItAsync` no need to reference Orchestration in it.